File: lib/models/blood_pressure.dart
class BloodPressure {
  final int id;
  final int userId;
  final DateTime testDate;
  final double bpLevel;
  final String? imageUrl;

  BloodPressure({
    required this.id,
    required this.userId,
    required this.testDate,
    required this.bpLevel,
    this.imageUrl,
  });

  // Factory constructor to create a BloodPressure instance from a map (JSON)
  factory BloodPressure.fromJson(Map<String, dynamic> json) {
    return BloodPressure(
      id: json['id'] as int,
      userId: json['user_id'] as int,
      testDate: DateTime.parse(json['test_date'] as String),
      bpLevel: json['bp_level'] as double,
      imageUrl: json['image_url'] as String?,
    );
  }

  // Method to convert a BloodPressure instance to a map (JSON)
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'test_date': testDate.toIso8601String(),
      'bp_level': bpLevel,
      'image_url': imageUrl,
    };
  }
}

File: models-kotlin/BloodPressure.kt
package com.healthtracker.models

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

data class BloodPressure(
    val id: Int,
    @SerializedName("user_id")
    val userId: Int,
    @SerializedName("test_date")
    val testDate: LocalDateTime,
    @SerializedName("bp_level")
    val bpLevel: Double,
    @SerializedName("image_url")
    val imageUrl: String? = null
) {
    companion object {
        private val dateFormatter = DateTimeFormatter.ISO_LOCAL_DATE_TIME

        fun fromJson(json: Map<String, Any?>): BloodPressure {
            return BloodPressure(
                id = json["id"] as Int,
                userId = json["user_id"] as Int,
                testDate = LocalDateTime.parse(json["test_date"] as String, dateFormatter),
                bpLevel = json["bp_level"] as Double,
                imageUrl = json["image_url"] as String?
            )
        }
    }

    fun toJson(): Map<String, Any?> {
        return mapOf(
            "id" to id,
            "user_id" to userId,
            "test_date" to testDate.format(dateFormatter),
            "bp_level" to bpLevel,
            "image_url" to imageUrl
        )
    }
}